### PR TITLE
[#158589222] Request services list to opt out

### PR DESCRIPTION
**Pull request by lmorelli986**
* [#158589222] - Add raw codebase in app, controller and service
* [#158589222] - Upgraded digital-citizenship library dependencies v0.44.0 > v0.47.0
* [#158589222] - Updated proxy api
* [#158589222] - build service
* [#158589222] - lint fix
* [#158589222] - Add tests
* [#158589222] - log errors in case of generic error responses
* [#158589222] - Adapt to snake case variable name "page_size"

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#158589222](https://www.pivotaltracker.com/story/show/158589222): Come APP voglio poter richiedere la lista dei servizi da cui CIT può fare opt-out

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/services/__tests__/messagesService.test.ts [hunk 1]
```diff
@@ -266,6 +297,58 @@ describe("MessageService#getMessagesByUser", () => {
...
+  it("returns an error if the getServicesByUser API returns an error", async () => {
+    mockGetServicesByUser.mockImplementation(() => {
+      return problemJson;
+    });
+
+    const service = new MessageService(api);
+
+    try {
+      await service.getServicesByUser(mockedUser);
+    } catch (e) {
```
Comment (gunzip) [inline src/services/__tests__/messagesService.test.ts]: if `getServicesByUser `does not throw the following expect calls would be skipped and the test will pass. you have to call `expect.assertions(3)` before the try / catch block to check that the three expectations will be called

File: src/services/__tests__/messagesService.test.ts [hunk 2]
```diff
@@ -266,6 +297,58 @@ describe("MessageService#getMessagesByUser", () => {
...
+    mockGetServicesByUser.mockImplementation(() => {
+      return invalidApiServicesResponse;
+    });
+
+    const service = new MessageService(api);
+
+    try {
+      await service.getServicesByUser(mockedUser);
+    } catch (e) {
+      expect(mockGetClient).toHaveBeenCalledWith(aValidFiscalCode);
```
Comment (gunzip) [inline src/services/__tests__/messagesService.test.ts]: ditto

File: api_proxy.yaml [hunk 3]
```diff
@@ -41,6 +41,39 @@ paths:
...
+                - service_id: 2b3e728c1a5d1efa035c
+        "401":
+          description: Bearer token null or expired.
+        "500":
+          description: There was an error in retrieving the services.
+          schema:
+            $ref: "#/definitions/ProblemJson"
+      parameters:
+        - $ref: "#/parameters/PaginationRequest"
+#        - name: recipient
```
Comment (cloudify) [inline api_proxy.yaml]: remove the commented lines

File: src/services/messagesService.ts [hunk 4]
```diff
@@ -185,4 +188,48 @@ export default class MessagesService {
...
+      log.error(
+        "Unknown response from getServicesByUser API: %s",
+        ReadableReporter.report(errorOrServices)
+      );
+      return left(internalError(messageErrorOnUnknownResponse));
+    }
+
+    const apiServices = errorOrServices.value;
+
+    // serve un to AppService che risolva restituendo solo ServiceId
```
Comment (cloudify) [inline src/services/messagesService.ts]: comment in English

File: src/services/messagesService.ts [hunk 5]
```diff
@@ -185,4 +188,48 @@ export default class MessagesService {
...
+    const appServices = apiServices.items; // .map(toAppService_X);
```
Comment (cloudify) [inline src/services/messagesService.ts]: is this still TODO/WIP?

File: src/services/__tests__/messagesService.test.ts (hunk 1 above)
Comment (lmorelli986) [reply src/services/__tests__/messagesService.test.ts]: Ok, I'll do it.

File: src/services/__tests__/messagesService.test.ts (hunk 2 above)
Comment (lmorelli986) [reply src/services/__tests__/messagesService.test.ts]: ditto

File: api_proxy.yaml (hunk 3 above)
Comment (lmorelli986) [reply api_proxy.yaml]: Oh yes I forgot to remove them, sorry.

File: src/services/messagesService.ts (hunk 4 above)
Comment (lmorelli986) [reply src/services/messagesService.ts]: ditto, because they are unnecessary

File: src/services/messagesService.ts (hunk 5 above)
Comment (lmorelli986) [reply src/services/messagesService.ts]: ditto

File: api_proxy.yaml [hunk 6]
```diff
@@ -442,6 +503,21 @@ definitions:
...
       - items
       - page_size
+  Services:
+    type: object
+    title: A list of services
+    properties:
+      items:
+        type: array
+        items:
+          $ref: "#/definitions/ServiceId"
```
Comment (cloudify) [inline api_proxy.yaml]: would it be possible to return the tuple `(serviceId, version)` ? this would make it easier for the app to cache the results

Comment (gunzip) [reply api_proxy.yaml]: yes

File: api_proxy.yaml [hunk 7]
```diff
@@ -41,6 +41,52 @@ paths:
...
+    get:
+      operationId: getVisibleServices
+      summary: Get all visible services
+      description:  |-
+        Returns the description of all visible services.
+      responses:
+        "200":
+          description: Found.
+          schema:
+            $ref: "#/definitions/VisibleServices"
```
[the remark this answers is not in the document]
Comment (gunzip) [reply api_proxy.yaml]: ok but the API (digital-citizenship-functions) actually returns the whole service object so we must change that one in the first place

[the remark this answers is not in the document]
Comment (cloudify) [reply api_proxy.yaml]: it's ok, I would just return `serviceId, version` to the app, discarding everything else

you may want to create a story for fixing the functions API later (but it's low priority)

File: api_proxy.yaml (hunk 6 above)
Comment (cloudify) [reply api_proxy.yaml]: ok pls add a story, we'll do it later, we'll use just the `serviceId` for now

File: api_proxy.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (gunzip) [reply api_proxy.yaml]: https://github.com/teamdigitale/digital-citizenship-functions/pull/330

File: api_proxy.yaml (hunk 6 above)
Comment (gunzip) [reply api_proxy.yaml]: https://github.com/teamdigitale/digital-citizenship-functions/pull/330

**Pull request merged**